database/src/error.rs

```diff
@@ -52,7 +52,7 @@ impl std::error::Error for DatabaseError {
         }
     }
 
-    fn cause(&self) -> Option<&std::error::Error> {
+    fn cause(&self) -> Option<&dyn std::error::Error> {
```

Comment:
Why the dyn here? meaning I thought that was only for traits, and I thought Error was a concrete